src/cmd-build

```diff
@@ -263,12 +276,30 @@ fi
 
 # And add the commit metadata itself, which includes notably the rpmdb pkglist
 # in a format that'd be easy to generate diffs out of for higher level tools
-"${dn}"/commitmeta_to_json "${workdir}/repo" "${commit}" > commitmeta.json
+"${dn}"/commitmeta_to_json "${tmprepo}" "${commit}" > commitmeta.json
+
+# And create a tarball of an ostree repo containing the commit
+if [ "${commit}" == "${previous_commit}" ]; then
+    cp -a --reflink=auto "${previous_builddir}/ostree-commit.tar" .
+else
+    ostree init --repo=repo --mode=archive
```

Comment:
I think we should keep around `cache/repo-archive`.  This way we don't pay the recompression penalty every build.

Reply:
Or hmm...if we teach anaconda to pull from the `bare-user` repo directly then we don't even need an archive repo at all, but I think that may run into issues with xattrs and 9p; we could also do a custom webserver that did dynamic `gzip -1` for objects and faked up being an archive repo.

Reply:
I think we should be good with this now. `pull-local` between archive repos just hardlinks (given that `tmp/` is on the same filesystem as `builds/`, which I think we unofficially assume right now).

Reply:
OK, let me sketch out the flow now with this latest revision:

- `rpm-ostree compose tree` commits into `tmp/repo`, so it performs compression there during the internal "`pull-local`" when importing from the temporary bare-user repo.
- by default, we keep the last 3 builds in `tmp/repo`, so during new composes we only pay for compression of new objects (and if the repo is wiped out, we reseed it from the last build).
- prepping the final `ostree-commit.tar` is a `pull-local` between two archive repos, so those are just hardlinks.